[Jonathan Wiguna] iP

JonathanWiguna opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/JonathanWiguna/ip/releases/download/v0.1/duke.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smile:

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("data/TaskList.txt").run();
    }
}
```

Commit:
Add Gradle support

JonathanWiguna committed:
Level 1

JonathanWiguna committed:
Level-2

JonathanWiguna committed:
Level-3

JonathanWiguna committed:
Level-4

JonathanWiguna committed:
A-TextUiTesting

JonathanWiguna committed:
Level-5

JonathanWiguna committed:
Level-6

JonathanWiguna committed:
Add functionality to read tasks stored in local storage

JonathanWiguna committed:
Removed print statements for tasks read from local storage

JonathanWiguna committed:
Create separate methods to create directory and to get tasks from local file

JonathanWiguna committed:
Add functionality to write added tasks into local file

JonathanWiguna committed:
Add functionality to write data into local storage

JonathanWiguna committed:
Change representation of date to LocalDateTime object

JonathanWiguna committed:
Merge branch 'branch-Level-7'

JonathanWiguna committed:
Resolve merge conflict in Deadline and Event

JonathanWiguna committed:
Restructure write task method

JonathanWiguna committed:
Add more OOP and add duke package

JonathanWiguna committed:
Refactor exceptions

JonathanWiguna committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

JonathanWiguna committed:
Add gradle support

JonathanWiguna committed:
Add some JUnit Tests

JonathanWiguna committed:
Add JavaDocs

JonathanWiguna committed:
Adhere to coding standard

JonathanWiguna committed:
Add find functionality

JonathanWiguna committed:
Merge branch 'branch-A-JavaDoc'

JonathanWiguna committed:
Merged branch-A-CodingStandard

JonathanWiguna committed:
Merge branch 'branch-Level-9'

JonathanWiguna committed:
Add CheckStyle to detect coding style violations

JonathanWiguna committed:
Merge branch 'branch-A-CheckStyle'

JonathanWiguna committed:
Fix left over coding style violation

JonathanWiguna committed:
Implement GUI using javafx

thamruicong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+     */
+    public int size() {
+        return this.tasks.size();
+    }
```
Would getSize() be a better name for this function?

thamruicong commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,126 @@
+
+    private static final String LINE = "__________________________________";
+    private static final String MESSAGE_WELCOME = "Hi there! Baymax at your service. Let me retrieve your stored task list!";
+    private static final String MESSAGE_BYE = "Goodbye!";
```
I like how you used screaming snake case for constants 😄

thamruicong commented on src/main/java/duke/commands/UnMarkCommand.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Represents an executable <code>Command</code> to mark a <code>Task</code> as not done.
+ */
+public class UnMarkCommand extends Command {
```
Perhaps UnmarkCommand instead of UnMarkCommand? 🤔

thamruicong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+     * @return <code>Command</code> to be executed.
+     * @throws DukeException If input is not of a recognizable pattern.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```
I really liked how you abstracted each command into its own class 👍. It makes your code easier to read.

thamruicong reviewed (commented):
Overall, I found your code to be really easy to follow and understand. Great job man! 💯

JonathanWiguna committed:
Delete Ui class and refactor rest of code

JonathanWiguna committed:
Merge branch 'branch-Level-10'

anuanas2007 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+            return new UnmarkCommand(fullCommand);
+        }
+        if (fullCommand.matches("^delete [0-9]*$")) {
+            return new DeleteCommand(fullCommand);
```
Perhaps you could consider using switch statements to make it more cleaner?

anuanas2007 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+     * @param index Index of the task.
+     * @return The <code>Task</code> in the specified index.
+     */
+    public Task get(int index) {
+        return this.tasks.get(index);
+    }
```
Perhaps you could handle exceptions for out of range indices?

anuanas2007 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+    public String list() {
+        Task[] x = new Task[tasks.size()];
+        Task[] tasksArray = tasks.toArray(x);
+        String response = "Here are the tasks in your list:\n";
+        for (int i = 1; i <= tasksArray.length; i++) {
+            Task task = tasksArray[i - 1];
+            response += i + "." + task.toString() + "\n";
+        }
```
Perhaps you could use Java `StringBuilder()` class to make it more efficient?

anuanas2007 commented on src/main/java/duke/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Represents an executable <code>Command</code> to add <code>Deadline</code>.
+ */
+public class AddDeadlineCommand extends Command {
```
Perhaps you could have one AddCommand for all event types?

anuanas2007 reviewed (commented):
Overall your code looks good.
Just a few suggestions!!
Maybe you can add more exception handling?

JonathanWiguna committed:
Add some assertions

JonathanWiguna committed:
Split a long method in the Storage class

The load() method in the Storage class is too long.
The parsing of the string read by the scanner into
a task can be done in another method.

Let's create another method to do the parsing.

JonathanWiguna committed:
Merge pull request #1 from JonathanWiguna/branch-A-Assertions

Add some assertions

JonathanWiguna committed:
Merge branch 'master' into branch-A-CodeQuality

JonathanWiguna committed:
Merge pull request #2 from JonathanWiguna/branch-A-CodeQuality

Split a long method in the Storage class

JonathanWiguna committed:
Fix bugs in the GUI

If the list of task is too long, it will be cut short on the next user
input. Also, the GUI window does not close when the user types "bye"

Let's,
* add a property to the DialogBox.fxml so that the DialogBox will
    not be cut short.
* add a 1 second delay before closing the window after the user types
    "bye"

JonathanWiguna committed:
Add .yml file for Continuous Integration

JonathanWiguna committed:
Add functionality to store recurring tasks

JonathanWiguna committed:
Merge branch 'branch-B-RecurringTasks'

JonathanWiguna committed:
Fix bug in CI

JonathanWiguna committed:
Fix some header comments

JonathanWiguna committed:
Change Baymax text to Duke

JonathanWiguna committed:
Fix response string for marking non-recurring task

JonathanWiguna committed:
Updated javadocs for RecurringTask

JonathanWiguna committed:
Add UI screenshot

JonathanWiguna committed:
Remove magic words and magic numbers

JonathanWiguna committed:
Add User Guide

JonathanWiguna committed:
Fix broken link

JonathanWiguna committed:
Fix broken formatting

JonathanWiguna committed:
Fix JAR file

JonathanWiguna committed:
Disable window resizing

JonathanWiguna committed:
Fix bug for the find command

JonathanWiguna committed:
Fix Duke image for Linux OS
